my_each.rb

```diff
@@ -1,6 +1,143 @@
 module Enumerable
-    def my_each(array)
-      for i in 0...array.length
-        p "gaa"
+  def my_each
+    (0...length).each do |i|
+      yield self[i]
```

Comment:
The purpose of the method is to create it from scratch. Please don't use the `each` method here.
Also, take into consideration that not all enumerables are arrays, so at some point, you will want to covert self into one to iterate over it.